builtin/aws/ecs/platform.go

```diff
@@ -1339,21 +1339,89 @@ func (p *Platform) resourceTargetGroupDestroy(
 		return nil
 	}
 
-	s := sg.Add("Deleting target group %s", state.Name)
+	s := sg.Add("Getting target group details", state.Name)
 	defer s.Abort()
 
 	elbsrv := elbv2.New(sess)
 
+	groups, err := elbsrv.DescribeTargetGroups(&elbv2.DescribeTargetGroupsInput{
+		TargetGroupArns: aws.StringSlice([]string{state.Arn}),
+	})
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to describe target group %s (ARN: %q): %s", state.Name, state.Arn, err)
+	}
+
+	s.Update("Retrieved target group details")
+	s.Done()
+
+	// There should be only one target group we "loop through" here
+	for _, group := range groups.TargetGroups {
```

Comment:
This is quite a big loop! I think it's worth considering if there's any way we could reduce this down from 5 nested for loops 😅